src/main/java/org/gridsuite/loadflow/server/service/FilterService.java

```diff
@@ -156,14 +197,55 @@ public List<ResourceFilter> getResourceFilters(@NonNull UUID networkUuid, @NonNu
 
         Network network = getNetwork(networkUuid, variantId);
 
-        List<String> subjectIdsFromEvalFilter = new ArrayList<>();
+        final List<AbstractFilter> genericFilters = getFilters(globalFilter.getGenericFilter());
+
+        EnumMap<EquipmentType, List<String>> subjectIdsByEquipmtType = new EnumMap<>(EquipmentType.class);
```

Comment:
subjectIdsByEquipmentType

Reply:
should it be better with Set ? To avoid dupes among same type of equipment ? can it happen ?